### ISSUE TITLE
4.0 Beta: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by franperezlopez** [careteditor/caret#592]
I've got this error when opening the preferences on the File -> Preferences. Also I've got a very similar error when trying to open any md file

The following error has occurred:
```
{
    "message": "Maximum call stack size exceeded",
    "version": "4.0.0-beta0",
    "os": "Windows 10",
    "path": "C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js",
    "code": "throw err0",
    "line": 90,
    "column": 21,
    "stack": [
        "at String.slice ((native)",
        "at Object.resolve ((path.js:290:29)",
        "at Object._makeLong ((path.js:664:32)",
        "at fs.mkdirSync ((fs.js:895:35)",
        "at Object.fs.mkdirSync ((ELECTRON_ASAR.js:669:16)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:71:13)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)"
    ]
}
```

btw, congratulations on delivering the new beta. the improvement in performance is really noticeable 

keep on the good work!!

**Comment by astoilkov** [comment]
Hi @franperezlopez,

Can you replicate the problem everytime you open `File -> Preferences`?

Thanks for the kind words.

**Comment by franperezlopez** [comment]
you're right .. I'm not able to find a pattern for the preferences dialog error ...

yet, I keep really consistent errors just opening files in the editor (via Folder treeview on the left), and always related with the stack size

```
{
    "message": "Maximum call stack size exceeded",
    "version": "4.0.0-beta0",
    "os": "Windows 10",
    "path": "C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js",
    "code": "throw err0",
    "line": 90,
    "column": 21,
    "stack": [
        "at String.slice ((native)",
        "at normalizeStringWin32 ((path.js:64:22)",
        "at Object.resolve ((path.js:304:20)",
        "at Object._makeLong ((path.js:664:32)",
        "at fs.mkdirSync ((fs.js:895:35)",
        "at Object.fs.mkdirSync ((ELECTRON_ASAR.js:669:16)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:71:13)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:77:24)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)",
        "at sync ((C:\\Users\\fran\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)"
    ]
}
```

**Comment by clockler** [comment]
Apparently this is caused by garbage paths on Windows - nonexistent drive letters reliably throw this, but I can't pin down any other causes (invalid characters throw `EPERM` instead). [`make-dir`](https://github.com/sindresorhus/make-dir) handles this better than `mkdirp` by actually validating paths and throwing `ENOENT` on a nonexistent drive letter, rather than apparently repeatedly trying until it exceeds the call stack.

**Comment by ramthir** [comment]
I too get the same error when opening any md file.

```json
{
    "message": "Maximum call stack size exceeded",
    "version": "4.0.0-beta0",
    "os": "Windows 10",
    "path": "C:\\Users\\ramesh.thiruchelvam\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js",
    "code": "throw err0",
    "line": 90,
    "column": 21,
    "stack": [
        "at String.slice ((native)",
        "at Object.resolve ((path.js:290:29)",
        "at Object._makeLong ((path.js:664:32)",
        "at fs.mkdirSync ((fs.js:895:35)",
        "at Object.fs.mkdirSync ((ELECTRON_ASAR.js:669:16)",
        "at sync ((C:\\Users\\ramesh.thiruchelvam\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:71:13)",
        "at sync ((C:\\Users\\ramesh.thiruchelvam\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:77:24)",
        "at sync ((C:\\Users\\ramesh.thiruchelvam\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)",
        "at sync ((C:\\Users\\ramesh.thiruchelvam\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)",
        "at sync ((C:\\Users\\ramesh.thiruchelvam\\AppData\\Local\\caret-beta\\app-4.0.0-beta0\\resources\\app.asar\\node_modules\\mkdirp\\index.js:78:17)"
    ]
}
```

**Comment by astoilkov** [comment]
This is now fixed in a new Caret 4.0 Beta update. You can download it from [here](https://caret.io/4beta/).

Thanks for all the additional information and suggestions.

Closing this issue.
